src/tbb/task_group_context.cpp

```diff
@@ -52,84 +52,12 @@ void tbb_exception_ptr::throw_self() {
 //------------------------------------------------------------------------
 
 void task_group_context_impl::destroy(d1::task_group_context& ctx) {
-    __TBB_ASSERT(!is_poisoned(ctx.my_owner), NULL);
-
-    auto ctx_lifetime_state = ctx.my_lifetime_state.load(std::memory_order_relaxed);
-    __TBB_ASSERT(ctx_lifetime_state != d1::task_group_context::lifetime_state::locked, nullptr);
-
-    if (ctx_lifetime_state == d1::task_group_context::lifetime_state::bound) {
+    __TBB_ASSERT(!is_poisoned(ctx.my_context_list), nullptr);
+    
+    if (ctx.my_lifetime_state.load(std::memory_order_relaxed) == d1::task_group_context::lifetime_state::bound) {
```

Comment:
Maybe check that `ctx.my_context_list` is not `nullptr` and assert that `ctx.my_lifetime_state.load(std::memory_order_relaxed) == d1::task_group_context::lifetime_state::bound`?